### ISSUE TITLE
BernardPHP connections hang when polling an empty queue

### ISSUE THREAD

**Issue by twispt** [cloudamqp/amqproxy#22]
What I've observed is when our BernardPHP / PhpAmqplib consumer is targeting the Amqproxy, it will hang when no messages remain in the queue. Essentially, it's not able to poll for messages and will hang at a variable point in time when the queue depth is zero. 

I put together a set of php files that reproduces the issue.
[bernard_test.tar.gz](https://github.com/cloudamqp/amqproxy/files/3498858/bernard_test.tar.gz)

**Comment by carlhoerberg** [comment]
Looks like you're using `AMQPSSLConnection` even when connecting to the proxy

**Comment by carlhoerberg** [comment]
When connecting to the proxy you need to use a plain text connection, `AMQPStreamConnection`

**Comment by carlhoerberg** [comment]
But on your problem description is sounds like you are able to connect and only when the queue is empty it hangs?

**Comment by twispt** [comment]
Yes, that is what it seems like. A few seconds after the queue becomes
empty, php hangs requiring a sig kill.

**Comment by twispt** [comment]
and, We are using AMQPSSLConnection, but with SSL Options set to null, which uses plain text and allow us to connect.

**Comment by twispt** [comment]
I also tried using AMQPStreamConnection, behavior is the same.  Still hangs when attempting to poll the empty queue.

**Comment by carlhoerberg** [comment]
Issue found, new release out soon

**Comment by carlhoerberg** [comment]
new 0.4.2 release out, should be fixed now
